ftw/contentpage/browser/newslisting.py

```diff
@@ -67,20 +69,41 @@ def get_portlet(self):
         if not manager_name or not name:
             return
 
-        manager = getUtility(
-            IPortletManager,
-            name=manager_name,
-            context=self.context)
-        assignments = getMultiAdapter(
-            (self.context, manager),
-            IPortletAssignmentMapping,
-            context=self.context)
-
-        if name in assignments:
-            return queryMultiAdapter(
-                (self.context, self.request, self,
-                 manager, assignments[name]),
-                IPortletRenderer)
+        context = self.context
+
+        # Prepare a list of objects by walking up the path.
+        contexts = [context]
+        while not IPloneSiteRoot.providedBy(context):
+            if IAcquirer.providedBy(context):
+                context = aq_parent(aq_inner(context))
+            else:
+                context = context.__parent__
```

Comment:
The parent pointer (``__pointer__``) is usually not stored in the ZODB. If an object has a parent pointer it is probably wrapped with an acquirer, therefore I don't really see a use case when the "else" is hit.
It could be something with assignments or other specially acquisition wrapped objects but I never had such a use case so I think it is safe to remove :wink: